accessibility-checker-engine/src/v2/checker/accessibility/util/legacy.ts

```diff
@@ -2069,6 +2069,20 @@ export class RPTUtil {
         return RPTUtil.hasInnerContentHiddenHyperLink(element, false);
     }
 
+    /**
+     * This function is responsible for checking if an element has an no empty text node as its first child.
+     *
+     * @parm {element} node The node checked.
+     * @return {bool} true if the element has an no empty text node as its first child, false otherwise
+     *
+     * @memberOf RPTUtil
+     */
+     public static hasNonEmptyTextNode(element) {
```

Comment:
Why a new function when we have `isInnerTextEmpty`?

Reply:
This particular function will return false for `<div role="menuitemradio"><span>Text</span></div>` or `<div role="menuitemradio">  <span>Text</span></div>`. Not considering all of the descendant nodes.

Reply:
the innerText or innerContent is the text from all the children and has noEmptyTextNode is the text from itself (my text), for example <div role='xxx'>my text <span>my span</span></div>.

Reply:
I think either a visible text in the div or an aria-label or aria-labelledby, aria-describredby should be fine.

Reply:
I'm not aware of any ARIA of HTML spec that only looks at the text nodes of the immediate children.